docs/docs/docs/01-core/configuration/00-intro.md

```diff
@@ -0,0 +1,115 @@
+
+### Base Config Names
+
+#### admin
+
+Defines settings related to the Strapi admin panel, such as the 'autoopen' option for `yarn develop`
+
+#### server
+
+Defines settings related to the Strapi backend server, such as 'host' and 'port'
+
+#### database
+
+Defines the database configuration options.
+
+NOTE: Most of this is defined by Knex, be careful when adding options that they are compatible or placed in the appropriate location
+
+#### api
+
+Content API configuraiton, for example the 'rest.maxLimit' option to set the maximum `limit` available in a query.
+
+#### features
+
+Feature flags for enabling and configuring future features that would be breaking changes and cannot be enabled by default, or experimental features that are not ready for public stable release.
+
+#### plugins
+
+Contains plugin configurations, with each root-level value the id of a plugin, for example, 'users-permissions'
+
+IMPORTANT: Plugins is an exception to the normal namespacing in config.get(). An example plugins.js file has this shape:
+
+```
+module.exports = () => ({
+  graphql: {
+    enabled: true,
+    config: {
+      endpoint: '/graphql',
+
+      defaultLimit: 25,
+      maxLimit: 100,
+
+      apolloServer: {
+        tracing: true,
+      },
+    },
+  },
+});
+```
+
+However, the `config` attribute is flattened and removed so that its values are accessible at, for example, `strapi.config.get('plugins.graphql.endpoint')`
+
+NOTE: Because this configuration contains user and plugin keys and Strapi does not yet provide a method to extend the definitions, this configuration is not strictly typed and will not be loaded from the environment variables
+
+#### middlewares
+
+The configuration of middlewares.
+
+NOTE: This is the only configuration that is currently an array rather than an object. It also very loosely typed. Future improvements are planned to strongly type all the built-in Strapi middlewares.
+
+### Typings
+
+Most of the base configurations have been typed, although they are not currently publicly available. They can be found in: `packages/core/types/src/types/core/config`
+
+Any time a new configuration option is added to Strapi, it must also be added to the types for the appropriate config file (and once the env var loading feature is available, a parser for it such as string, integer, stringArray, etc must be defined)
+
+In Strapi v5, they will be used for:
+
+- user-facing config factories to assist in typing project configuration
+- as part of the environment variable configuration loading to validate the structure of the base config file parsers to parse and strongly type values from the environment
+
+### Configuration filename restrictions
+
+In Strapi v5, environment variables with the prefix `STRAPI_` will be loaded automatically intro Strapi configuration. Because of that, some naming restrictions have been added to prevent conflicts.
```

Comment:
Just a typo

```suggestion
In Strapi v5, environment variables with the prefix `STRAPI_` will be loaded automatically into Strapi configuration. Because of that, some naming restrictions have been added to prevent conflicts.
```